enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -894,6 +894,13 @@ func (s *ExecutionServer) updateUsage(ctx context.Context, cmd *repb.Command, ex
 	}
 	dur, err := executionDuration(executeResponse.GetResult().GetExecutionMetadata())
 	if err != nil {
+		// If the task encountered an error, it's somewhat expected that the
+		// execution duration will be unset, so don't return an error. For
+		// example, we may have failed to pull the image, so execution could not
+		// even begin.
+		if execErr := gstatus.ErrorProto(executeResponse.GetStatus()); execErr != nil {
```

Comment:
Are there cases where a status would not be OK and we'd still be able to compute the duration? 
i.e. why does this check need to be inside the error check and not before trying to compute duration?

Reply:
Added a comment - if a task timed out then I believe we'll still get an exec duration, and we still want to collect usage data for those.